src/ariel-os-hal/src/dummy/peripheral.rs

```diff
@@ -0,0 +1,16 @@
+pub use embassy_hal_internal::Peripheral;
+#[doc(hidden)]
+/// Dummy type.
+///
+/// See the `OptionalPeripherals` type of your Embassy HAL crate instead.
```

Comment:
optional: usually we keep the attributes as close as possible to the item (but it was indeed in this state before this refactor)

```suggestion
/// Dummy type.
///
/// See the `OptionalPeripherals` type of your Embassy HAL crate instead.
#[doc(hidden)]
```